### PR TITLE
base model class & comparison visualizer

### PR DESCRIPTION
**Pull request by m-bain**
Add a base_model class + heavy refactor of duplicate code in model generation scripts.

Also creates a visualizer for comparing two model generations + their ratings
Screenshot 2024-12-17 at 16 42 04

### CODE REVIEW

**Comment by m-bain** [comment]
@prazek

File: models/xai_models.py [hunk 1]
```diff
@@ -1,73 +1,51 @@
 """
-$ pip install openai # we use openai api sdk for simplicity
+$ pip install openai # we use openai sdk for simplicty
```
Comment (prazek) [inline models/xai_models.py:2]: maybe reword to "Xai uses openai api for simplicity" - this is a bit ambiguous, I firstly thought we do it for simplicity and for some reason it works

**Review by prazek — APPROVED** [review]
Thanks Max, this looks nice!
